### PR TITLE
UI: Make "Start streaming" and "Start recording" buttons checkable

### PR DESCRIPTION
**Pull request by nleseul**
Per a suggestion in the forum: https://obsproject.com/forum/threads/suggestion-color-code-the-streaming-and-recording-toggle-buttons.75209/

This makes these buttons stand out more visually when active, and also allows them to be skinned differently when active.

### CODE REVIEW

**Comment by Fenrirthviti** [comment]
In addition to the change to the buttons, you might want to include a screenshot of what this change looks like when paired with an updated theme (and probably update the themes themselves to utilize this), as this change alone does nothing to help the actual issue.

File: UI/forms/OBSBasic.ui [hunk 1]
```diff
@@ -978,6 +978,9 @@
        <property name="text">
         <string>Basic.Main.StartRecording</string>
        </property>
+      <property name="checkable">
```
Comment (RytoEX) [inline UI/forms/OBSBasic.ui]: This indentation appears to be missing one space.

**Comment by SuslikV** [comment]
The buttons has more than two states: Start/Connecting/Stopping/Stop (some has 3, some 4).

was mentioned in https://github.com/jp9000/obs-studio/pull/856

**Comment by nleseul** [comment]
@Fenrirthviti, here are some shots of how this looks with the built-in themes, with no other changes. The default QButton styling already knows how to render differently in a checked state, as seen in the existing "Studio Mode" button. I'd argue that the scope of this change really doesn't need to go any further than making the streaming and recording buttons behave in the same way as the "Studio Mode" button. 

Default theme: 
image

Dark theme: 
image

Rachni theme: 
image

Now, I do note that the "Rachni" theme seems like it has a unique checked style applied to the "Studio Mode" button that isn't getting applied to the other buttons with my change. I'd definitely like to look into getting that applied to the streaming and recording buttons too. 

@RytoEX, good catch. I've added that missing space. 

@SuslikV, thanks for pointing out #856. Regarding @jp9000's comment on closing that one, I think this change is much smaller in scope than that one and more reasonable as an interim approach, pending a larger overhaul of the UI. And while the button text does present four states, using the built-in two-state checkable behavior of QButton makes it much simpler to implement, and I think it gives the user sufficient visual feedback to be worthwhile (even if it doesn't immediately distinguish between "connecting" and "actively streaming" states).

**Comment by SuslikV** [comment]
Reconnecting can happen if you connection fails. And it can happen more often than you think. Sometimes people use laptops and Wi-Fi just to stream camera source... My suggestion was about free to modify your custom themes (if any). Later, there was Stats window implemented, where you can monitor tasks state (running/idle). Stats window still missing replay buffer for monitoring, of course, but it clearly indicates current status and "health" of the running task. Stats window was more important then themes at that time. And it more useful now days. I think, the decision to not deviate in favor of themes development was right. But now we can return to the scope of the UI development.

Also, from my point of view the `window-basic-main.cpp` s the most edited part of the project by the community. This means, that it has controversial basis. And your suggestion also acceptable.

**Comment by nleseul** [comment]
Here's Rachni updated to apply its custom "checked" styling to these buttons: 
image

I added the other main menu buttons to that group receiving this styling too, even though there's probably no reason they'll ever need to be "checked", just because it seems silly to me to have the styling inconsistent.

**Comment by Fenrirthviti** [comment]
Thanks for this. I did miss a few things when working on the Rachni theme, thanks for cleaning that one up!

Looks great, very simple QoL improvement.

**Comment by jp9000** [comment]
I do like this PR quite a bit better than other proposals that have come before.  I'll accept this one.  Make sure to squash your commits however next time.  I'm going to squash your commits, then merge your changes.  You should see "closed in [commit hash]".

**Pull request closed without merging**